test/unit/math/mix/scal/fun/log_diff_exp_test.cpp

```diff
@@ -1,358 +1,29 @@
-                  g[0]);
-}
-
-TEST(AgradFwdLogDiffExp, Double_FvarFvarVar_2ndDeriv) {
-  using stan::math::fvar;
-  using stan::math::log_diff_exp;
-  using stan::math::var;
-  using std::exp;
-
-  double x(9.0);
-  fvar<fvar<var> > y;
-  y.val_.val_ = 6.0;
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > a = log_diff_exp(x, y);
-
-  AVEC p = createAVEC(y.val_.val_);
-  VEC g;
-  a.d_.val_.grad(p, g);
-  EXPECT_FLOAT_EQ((-exp(6.0) * (exp(9.0) - exp(6.0)) + exp(6.0) * -exp(6.0))
-                      / (exp(9.0) - exp(6.0)) / (exp(9.0) - exp(6.0)),
-                  g[0]);
-}
-TEST(AgradFwdLogDiffExp, FvarFvarVar_FvarFvarVar_3rdDeriv) {
-  using stan::math::fvar;
-  using stan::math::log_diff_exp;
-  using stan::math::var;
-  using std::exp;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 9.0;
-  x.val_.d_ = 1.0;
-
-  fvar<fvar<var> > y;
-  y.val_.val_ = 6.0;
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > a = log_diff_exp(x, y);
-
-  AVEC p = createAVEC(x.val_.val_, y.val_.val_);
-  VEC g;
-  a.d_.d_.grad(p, g);
-  EXPECT_FLOAT_EQ(-0.060919307, g[0]);
-  EXPECT_FLOAT_EQ(0.060919307, g[1]);
-}
-TEST(AgradFwdLogDiffExp, FvarFvarVar_Double_3rdDeriv) {
-  using stan::math::fvar;
-  using stan::math::log_diff_exp;
-  using stan::math::var;
-  using std::exp;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 9.0;
-  x.val_.d_ = 1.0;
-  x.d_.val_ = 1.0;
-
-  double y(6.0);
-
-  fvar<fvar<var> > a = log_diff_exp(x, y);
-
-  AVEC p = createAVEC(x.val_.val_);
-  VEC g;
-  a.d_.d_.grad(p, g);
-  EXPECT_FLOAT_EQ(0.060919308279076959008006310, g[0]);
-}
-
-TEST(AgradFwdLogDiffExp, Double_FvarFvarVar_3rdDeriv) {
-  using stan::math::fvar;
-  using stan::math::log_diff_exp;
-  using stan::math::var;
-  using std::exp;
-
-  double x(9.0);
-  fvar<fvar<var> > y;
-  y.val_.val_ = 6.0;
-  y.d_.val_ = 1.0;
-  y.val_.d_ = 1.0;
-
-  fvar<fvar<var> > a = log_diff_exp(x, y);
-
-  AVEC p = createAVEC(y.val_.val_);
-  VEC g;
-  a.d_.d_.grad(p, g);
-  EXPECT_FLOAT_EQ(-0.060919308279076959008006309952, g[0]);
-}
-
-struct log_diff_exp_fun {
-  template <typename T0, typename T1>
-  inline typename boost::math::tools::promote_args<T0, T1>::type operator()(
-      const T0 arg1, const T1 arg2) const {
-    return log_diff_exp(arg1, arg2);
-  }
-};
-
-TEST(AgradFwdLogDiffExp, nan) {
-  log_diff_exp_fun log_diff_exp_;
-  test_nan_mix(log_diff_exp_, 3.0, 5.0, false);
+#include <test/unit/math/test_ad.hpp>
+#include <limits>
+
+TEST(mathMixScalFun, logDiffExp) {
+  auto f = [](const auto& x1, const auto& x2) {
+    return stan::math::log_diff_exp(x1, x2);
+  };
+  // stan::test::expect_common_nonzero_binary(f);
+  stan::test::expect_ad(f, 1.2, 0.5);
+  stan::test::expect_ad(f, 5.0, 2.0);
+  stan::test::expect_ad(f, 9.0, 6.0);
+  stan::test::expect_ad(f, 0.1, 0.0);
+  stan::test::expect_ad(f, 3.01, 2.0);
+  stan::test::expect_ad(f, 2.0, 1.0);
+
+  // hessians near zero have high relative error but small absolute error
+  stan::test::ad_tolerances tols;
```

Comment:
If we decide to document these, here's one.

Reply:
If we want to doc these may be good to add to the autodiff page of the new stan math site I'm setting up

https://github.com/stan-dev/math/pull/1397